builtin/credential/aws/path_role_test.go

```diff
@@ -986,6 +986,65 @@ func TestAwsVersion(t *testing.T) {
 	}
 }
 
+// This test was used to reproduce https://github.com/hashicorp/vault/issues/7418
+// and verify its fix.
+func TestRoleResolutionWithSTSEndpointConfigured(t *testing.T) {
+	t.Skip("skipping test because it hits real endpoints")
+
+	config := logical.TestBackendConfig()
+	storage := &logical.InmemStorage{}
+	config.StorageView = storage
+
+	b, err := Backend(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = b.Setup(context.Background(), config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// configure the client with an sts endpoint that should be used in creating the role
+	data := map[string]interface{}{
+		"sts_endpoint": "https://sts.eu-west-1.amazonaws.com",
+		// Note - if you comment this out, you can reproduce the error shown
+		// in the linked GH issue above. This essentially reproduces the problem
+		// we had when we didn't have an sts_region field.
+		"sts_region": "eu-west-1",
```

Comment:
Strictly speaking, this might still work if the client happened to pick eu-west-1, right? So probably worth it to leave a comment stating that this test should be run 2-3 times to make sure it's really solved?